packages/@uppy/image-generator/src/index.tsx

```diff
@@ -0,0 +1,377 @@
+    this.unmount()
+  }
+
+  private closeAssembly(cancel = false): void {
+    if (this.assembly) {
+      const { status } = this.assembly
+      this.assembly.close()
+      this.assembly = null
+
+      // Cancel the assembly on the server to stop processing
+      if (cancel && status) {
+        this.client.cancelAssembly(status).catch(() => {
+          // If we can't cancel, there's not much we can do
+        })
+      }
+    }
+  }
+
+  private clearLoadingInterval(): void {
+    if (this.loadingInterval) {
+      clearInterval(this.loadingInterval)
+      this.loadingInterval = null
+    }
+  }
+
+  private startLoadingAnimation(): void {
+    this.clearLoadingInterval()
+    this.loadingInterval = setInterval(() => {
+      const { loadingMessageIndex } = this.getPluginState()
+      const nextIndex = (loadingMessageIndex + 1) % LOADING_MESSAGES.length
+      this.setPluginState({ loadingMessageIndex: nextIndex })
+    }, 4000)
+  }
+
+  /**
+   * Creates a Transloadit assembly to generate AI images.
+   *
+   * Completion scenarios:
+   * - Success: assembly emits 'finished' → resolve() → finally cleans up, keeps results
+   * - Error: assembly emits 'error' → reject() → catch reports error, finally cleans up
+   * - Dashboard close: onDashboardClose sets cancelled=true, resolve() → finally resets state
+   * - Uninstall: closeAssembly(true) called directly, cancels server-side assembly
+   */
+  generate = async () => {
+    const { promise, resolve, reject } = Promise.withResolvers<void>()
+    let cancelled = false
+
+    const onDashboardClose = () => {
+      cancelled = true
+      resolve()
+    }
+
+    // @ts-expect-error not typed because we do not depend on @uppy/dashboard
+    this.uppy.once('dashboard:close-panel', onDashboardClose)
+
+    try {
+      this.setPluginState({
+        loading: true,
+        results: [],
+        checkedResultIds: new Set(),
+        loadingMessageIndex: 0,
+      })
+      this.startLoadingAnimation()
+
+      const assemblyOptions = await this.opts.assemblyOptions(
+        this.getPluginState().prompt,
+      )
+
+      const assemblyResponse = await this.client.createAssembly({
+        params: assemblyOptions.params,
+        fields: assemblyOptions.fields ?? {},
+        signature: assemblyOptions.signature,
+        expectedFiles: 0,
+      })
+
+      const assembly = new Assembly(assemblyResponse, this.rateLimitedQueue)
+      this.assembly = assembly
+
+      assembly.on('result', (stepName: string, result: AssemblyResult) => {
+        const { results } = this.getPluginState()
+        this.setPluginState({
+          results: [...results, result],
+          firstRun: false,
+        })
+      })
+
+      assembly.on('error', reject)
+      assembly.on('finished', resolve)
+      assembly.connect()
+
+      await promise
+    } catch (error) {
+      this.client.submitError(error as Error).catch(() => {})
+      throw error
```

Comment:
shouldn't we inform the user ( visually ) that an error has occured ?  `uppy.info` ?